.github/workflows/push.yml

```diff
@@ -0,0 +1,20 @@
+name: Build and Test
+on: push
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v1
+      - run: sudo apt-get install -y valgrind
+      - run: pip3 install --user gcovr
+      - run: pip3 install --user ninja
+      - run: pip3 install --user meson
+      - uses: actions/checkout@v2
+      - run: meson setup build
+      - run: meson compile -C build
+      - run: ./build/test/sim/simtest
+      - run: gcovr --root build/src --xml-pretty -o coverage.xml
+      - uses: codecov/codecov-action@v2
+        with:
+          files: ./coverage.xml
+      - run: valgrind -v --error-exitcode=1 --leak-check=full --show-leak-kinds=all --track-origins=yes ./build/test/sim/simtest
```

Comment:
valgrind is an advisory failure and should not fail the build since it has gratuitous failures on threaded code.